src/Modules/UngradedModules/FreeMod.jl

```diff
@@ -153,13 +153,15 @@ end
 @doc raw"""
     rank(F::FreeMod)
     number_of_generators(F::AbstractFreeMod)
-    dim(F::AbstractFreeMod)
+    dim(F::AbstractFreeMod{T}) where T <: FieldElem
+    vector_space_dim(F::AbstractFreeMod{T}) where T <: FieldElem
 
 Return the rank of `F`.
 """
-dim(F::AbstractFreeMod) = rank(F)
 rank(F::FreeMod) = F.n
+dim(F::AbstractFreeMod{T}) where T <: FieldElem = rank(F)
```

Comment:
My understanding is that the removed method `dim(F::AbstractFreeMod)` is the source of many (most? all?) failures on this PR.

So I suggest we add it back temporarily, so we can get this PR merged. Then in a next step we can work in a ne PR on removing the method -- but in the meantime people can already start using `krull_dim` and `vector_space_dim`

```suggestion
dim(F::AbstractFreeMod{T}) where T <: FieldElem = rank(F)
dim(F::AbstractFreeMod) = rank(F)  # FIXME: remove this again
```